network/network_common.go

```diff
@@ -15,9 +15,63 @@ func (self *Network) Name() string {
 
 func (self *Network) Collect() (result interface{}, err error) {
 	result, err = getNetworkInfo()
```

Comment:
In order to not have `err` & `err2`, you should just return as soon as you encounter an error.

Since your change in `getMultiNetworkInfo()`, we now properly know when it fails (when it returns an error). However, I think that `Collect()` should be resilient to a fail in `getMultiNetworkInfo()`, because we still have enough data to function properly. 

Thus, I think we should just not do anything if the call to `getMultiNetworkInfo()` failed, and because it always return a valid slice, which can be empty, we could probably simplify the code this way:

```
func (self *Network) Collect() (result interface{}, err error) {
	result, err = getNetworkInfo()
	if err != nil {
		return
	}

	interfaces, err := getMultiNetworkInfo()
	if err == nil && len(interfaces) > 0 {
		interfaceMap, ok := result.(map[string]interface{})
		if !ok {
			return
		}
		interfaceMap["interfaces"] = interfaces
	}
	return
}
```

WDYT?

Reply:
No issues here. will commit